python/cudf/cudf/core/column/categorical.py

```diff
@@ -733,7 +733,7 @@ def categories(self) -> ColumnBase:
 
     @categories.setter
     def categories(self, value):
-        self.dtype = CategoricalDtype(
+        self._dtype = CategoricalDtype(
```

Comment:
Why this change?

Reply:
Because of line 1280 below. `CategoricalColumn.categories` setter was an un-reachable code until this PR which actually hid a bug here where `.dtype` has no setter in `ColumnBase`/`CategoricalColumn`.